react-graph/src/__tests__/Plot.test.ts

```diff
@@ -282,4 +282,16 @@ test('GetData with empty data', () => {
   const node = new PointNode([]);
   const points = node.GetData(0 ,10);
   expect(points.toString()).toBe([].toString());
-})
\ No newline at end of file
+})
+
+test('trimTree fundamental test', () => {
+  const data: [number, number][] = Array.from({length: 2000}, (_, i) => [i, i]);
+  const node = new PointNode(data);
+  expect(node.count).toBe(2000);
+  
+  node.AddPoints([2000, 2000]);
+  
+  const fullData = node.GetFullData();
+  expect(fullData.length).toBeLessThanOrEqual(1001);
```

Comment:
See above. While it does pass the test the test itself is somewhat flawed.
If a node can hold 2000 Points why would adding 1 point reesult in it holding only 1000 points now?

The better way to approach it is to keep it at 2000 points - so adding one point will remove one point (at the other end)
So realistically what we need is a `RemovePoints(n)` function instead a `trimtree` which would remove `n` points from the left most node